internal/mesh/internal/controllers/gatewayproxy/builder/builder.go

```diff
@@ -117,11 +117,7 @@ func (b *proxyStateTemplateBuilder) buildListener(address *pbcatalog.WorkloadAdd
 func (b *proxyStateTemplateBuilder) routers() []*pbproxystate.Router {
 	var routers []*pbproxystate.Router
 
-	if b.exportedServices == nil {
-		return routers
-	}
-
-	for _, exportedService := range b.exportedServices.Data.Services {
+	for _, exportedService := range b.exportedServices {
```

Comment:
So we don't need the nil check anymore because we will always have the `BlackHoleClusters` now?

Reply:
We don't need the nil check anymore because we loop over the property directly instead of accessing it via its parent

```go
for _, item := range items
```
instead of

```go
for _, item := range <*Parent>.items
```